src/Cli/dotnet/Commands/Solution/Add/SolutionAddCommand.cs

```diff
@@ -39,7 +38,7 @@ private static bool IsSolutionFolderPathInDirectoryScope(string relativePath)
 
     public SolutionAddCommand(ParseResult parseResult) : base(parseResult)
     {
-        _fileOrDirectory = parseResult.GetValue(SolutionCommandParser.SlnArgument);
+        _fileOrDirectory = parseResult.GetValue(SolutionCommandParser.SlnArgument)!;
```

Comment:
The `ParseResult` type in general does not work well with nullable analysis because `GetValue` is hard coded to return `T?`. One possible solution is add a `GetValueOrThrow` extension method. Happy to make that change if it's desired.

Reply:
We chatted about this briefly but the main gap here is that the S.CL library enforces the _arity_ of an option/argument but there's no way for that Arity to influence the return type of this method.